src/server.cpp

```diff
@@ -740,6 +740,15 @@ Response InternalServer::handle_search(const RequestContext& request)
   } catch (const std::exception& e) {
     std::cerr << e.what() << std::endl;
   }
+
+  // changing status code if no results obtained
+  searcher.restart_search();
+  
+  if(!searcher.getNextResult())
+  {
+    response.set_code(MHD_HTTP_NO_CONTENT);
+  }
```

Comment:
With the `restart_search` you are doing two searches.
One to generate the http content, one to test if there is results.

You can use `getEstimatedResultCount();` to check if there are result or not.

Reply:
ok i would include that changes as well in the next commit thanks

Reply:
ok I will include that change as well in next commi.....btw i thought of that earlier but didn't understood  meaning of estimate is that a thing related to upper bound?